apps/serdes/Deserializer.h

```diff
@@ -0,0 +1,32 @@
+#ifndef HALIDE_DESERIALIZER_H
+#define HALIDE_DESERIALIZER_H
+
+#include <string>
+#include <vector>
+
+#include "halide_ir_generated.h"
+#include <Halide.h>
+using namespace Halide;
+
+class Deserializer {
+public:
+    Deserializer() = default;
+
+    Pipeline deserialize(const std::string &filename);
+
+private:
+    // helper functions to deserialize each type of object
+    std::string deserialize_string(const flatbuffers::String *str);
+
+    Halide::Type deserialize_type(const Halide::Serdes::Type *type);
```

Comment:
I assume that `Serdes` is short for `Serialize/Deserialize`, but, uh, it's not a great name. I'd vote for `Serialize`; it seems pretty obvious that deserialization is included. But I'm open for other suggestions.

Reply:
I agree, `Serialize` reads good